### PR TITLE
support gpt-4o

### PR DESCRIPTION
**Pull request by leo4life2**

### CODE REVIEW

**Comment by rhlkmth** [comment]
thank you for this, i updated this in my private fork. Its a pleasure to use.

**Comment by jinjianming** [comment]
Now there's a problem with the compilation
=># info There appearances to be troublesome with your network connection Retrying
=># info There appearances to be troublesome with your network connection Retrying

File: app/utils.ts [hunk 1]
```diff
@@ -290,16 +290,13 @@ export function getMessageImages(message: RequestMessage): string[] {
...
     "vision",
     "claude-3",
     "gemini-1.5-pro",
+    "gpt-4-turbo",
+    "gpt-4o",
   ];
 
-  const isGpt4Turbo = model.includes("gpt-4-turbo") && !model.includes("preview");
-
-  return visionKeywords.some((keyword) => model.includes(keyword)) || isGpt4Turbo;
```
Comment (fred-bf) [inline app/utils.ts:304]: gpt-4-turbo still do not support vision mode

File: app/utils.ts [hunk 2]
```diff
@@ -290,16 +290,13 @@ export function getMessageImages(message: RequestMessage): string[] {
...
-
   const visionKeywords = [
     "vision",
     "claude-3",
     "gemini-1.5-pro",
+    "gpt-4-turbo",
+    "gpt-4o",
   ];
 
-  const isGpt4Turbo = model.includes("gpt-4-turbo") && !model.includes("preview");
```
Comment (fred-bf) [inline app/utils.ts:302]: gpt-4-turbo-preview still do not support vision mode

**Review by fred-bf — CHANGES_REQUESTED** [review]

**Comment by 04041b** [comment]
> Now there's a problem with the compilation =># info There appearances to be troublesome with your network connection Retrying =># info There appearances to be troublesome with your network connection Retrying

delete yarn.lock, there's problem with npm mirror

**Comment by leochen12-rgb** [comment]
问一下，基于api部署的gpt-4o官方是否收费

**Pull request merged**